rotkehlchen/exchanges/exchange.py

```diff
@@ -230,14 +230,31 @@ def query_online_income_loss_expense(
         Has to be implemented by exchanges if they have anything exchange specific
 
         For example coinbase
+
+        Deprecated, will be replaced by query_online_history_events
         """
-        raise NotImplementedError(
-            'query_online_income_loss_expense should only be implemented by subclasses',
-        )
+        return []
 
-    def query_history_events(self) -> None:
+    def query_online_history_events(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> Sequence['HistoryBaseEntry']:
+        """Queries the exchange's API for history events of the user
+
+        Should be implemented in subclasses.
+        Returns events based on HistoryBaseEntry, such as HistoryEvent, AssetMovement, etc.
+        """
+        # Should raise NotImplementedError here after all exchanges are updated
+        return []
+
+    def query_other_history_events(self) -> None:
```

Comment:
I saw what you wrote in the PR description but you should not rename old function and use a new function.

Should be combined.

This will now be called for each exchange the moment you enter the history view. It needs to be combined.

Reply:
Combined it. Now exchanges that need this will simply reimplement `query_history_events` with custom logic.